docs/src/developers_guide/contributing_tests.rst

```diff
@@ -0,0 +1,301 @@
+
+All unit tests for :py:class:`iris.cube.Cube` reside in:
+
+    :literal:`lib/iris/tests/unit/cube/test_Cube.py`
+
+Within that file the tests might look something like:
+
+.. code-block:: python
+
+    # Tests for the Cube.xml() method.
+    class Test_xml(tests.IrisTest):
+        def test_some_general_stuff(self):
+            ...
+
+
+    # Tests for the Cube.xml() method, focussing on the behaviour of
+    # the checksums.
+    class Test_xml__checksum(tests.IrisTest):
+        def test_checksum_ignores_masked_values(self):
+            ...
+
+
+    # Tests for the Cube.add_dim_coord() method.
+    class Test_add_dim_coord(tests.IrisTest):
+        def test_normal_usage(self):
+            ...
+
+        def test_coord_already_present(self):
+            ...
+
+
+.. _pytesting.functions:
+
+Functions
+---------
+
+When testing a function all the tests must reside in the module:
+
+    :literal:`lib/iris/tests/unit/<fully/qualified/module>/test_<function_name>.py`
+
+Within this test module there may be test classes, for example:
+
+* ``Test``
+* ``TestAspectOfFunction``
+
+Within those test classes, the test methods must be named according
+to the aspect of the tested function which they address.
+
+**Examples**:
+
+All unit tests for :py:func:`iris.experimental.raster.export_geotiff`
+must reside in:
+
+    :literal:`lib/iris/tests/unit/experimental/raster/test_export_geotiff.py`
+
+Within that file the tests might look something like:
+
+.. code-block:: python
+
+    # Tests focussing on the handling of different data types.
+    class TestDtypeAndValues(tests.IrisTest):
+        def test_int16(self):
+            ...
+
+        def test_int16_big_endian(self):
+            ...
+
+
+    # Tests focussing on the handling of different projections.
+    class TestProjection(tests.IrisTest):
+        def test_no_ellipsoid(self):
+            ...
+
+
+.. _pytesting.integration:
+
+Integration Tests
+-----------------
+
+Some code changes may require tests which exercise several units in
+order to demonstrate an important consequence of their interaction which
+may not be apparent when considering the units in isolation.
+
+These tests must be placed in the ``lib/iris/tests/integration`` folder.
+Unlike unit tests, there is no fixed naming scheme for integration
+tests. But folders and files must be created as required to help
+developers locate relevant tests. It is recommended they are named
+according to the capabilities under test, e.g.
+``metadata/test_pp_preservation.py``, and not named according to the
+module(s) under test.
+
+.. _pytesting_style_guide:
+
+PyTest Style Guide
+==================
+
+This style guide should be approached pragmatically. Most of the guidelines laid out
+below will not be practical in every scenario, and as such should not be considered
+firm rules.
+
+At time of writing, some existing tests have already been written in PyTest,
+so might not be abiding by these guidelines.
+
+Directory
+---------
```

Comment:
This section overlaps significantly with:

https://github.com/SciTools/iris/blob/37a342f3c2f005e945347ed0d58654c72fbc2b3c/docs/src/developers_guide/contributing_tests.rst?plain=1#L11-L12

Reply:
Yeah I think we can remove the ``Directory`` section